src/js/components/utils/editModal.tsx

```diff
@@ -3,36 +3,90 @@ import Input from '@salesforce/design-system-react/components/input';
 import Modal from '@salesforce/design-system-react/components/modal';
 import Textarea from '@salesforce/design-system-react/components/textarea';
 import i18n from 'i18next';
-import React, { useCallback, useEffect, useRef } from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 
-import { useForm } from '@/components/utils';
+import { useForm, useIsMounted } from '@/components/utils';
 import { Project } from '@/store/projects/reducer';
 
 interface EditModalProps {
   project: Project;
   isOpen: boolean;
   toggleModal: (open: boolean) => void;
+  handleClose: () => void;
 }
-const EditModal = ({ project, isOpen, toggleModal }: EditModalProps) => {
+const EditModal = ({
+  project,
+  isOpen,
+  toggleModal,
+  handleClose,
+}: EditModalProps) => {
+  const [saveBlocked, setSaveBlocked] = useState(true);
+  const isMounted = useIsMounted();
   const form = useRef<HTMLFormElement | null>(null);
   const submitButton = useRef<HTMLButtonElement | null>(null);
 
-  const { handleSubmitFromRef } = useForm({
-    fields: { name: '', description: '' },
+  const handleSuccess = () => {
+    setSaveBlocked(true);
+    handleClose();
+  };
+
+  const handleError = () => {
+    if (isMounted.current) {
+      setSaveBlocked(true);
+    }
+  };
+
+  const defaultName = project.name;
+  const defaultDescription = project.description;
+
+  const {
+    inputs,
+    errors,
+    handleInputChange,
+    setInputs,
+    handleSubmit,
+    resetForm,
+  } = useForm({
+    fields: {
+      name: defaultName,
+      description: defaultDescription,
+    },
+    additionalData: project,
```

Comment:
Hm, this isn't going to work quite right, because right now the data in `useForm` is being sent as `{ ...inputs, ...additionalData }`, so the old `project` data will override the changed data. We could probably fix that most easily in `useForm`...

Reply:
Alternatively, we could also do:

```suggestion
    fields: {
      name: defaultName,
      description: defaultDescription,
    },
    additionalData: _.omit(project, ['name', 'description']),
```